docs/src/token-swap.md

```diff
@@ -301,3 +301,39 @@ to explain fuzzing, but the specific implementation for the program can be found
 in the [instruction fuzz
 tests](https://github.com/solana-labs/solana-program-library/blob/master/token-swap/program/fuzz/src/instructions.rs)
 of the repo.
+
+## Account Diagram
+
+Below is a diagram showing all the accounts used by the Token Swap program. 
+
+![](../static/img/token-swap.png)
+
+And here's a quick rundown of how things work, using the names in the diagram as references.
+
+### Initialization
+
+- **Swap** is created.
+- **Authority** is verified. There is one **Authority** per **Swap**, and it should be a PDA account that uses the **Swap**'s public key as a seed. Also, it should own **TokenA** and **TokenB**, but not **PoolToken** and **FeeToken**. It needs to own the first two accounts in order to perform transfers.
+- An initial amount of pool tokens are minted into **PoolToken** by **Authority**.
+
+### Swap
+
+Let's say the desired swap is token A ⇾ token B.
+
+First, the user that wants to perform the swap creates two new token accounts for **TokenAMint** and **TokenBMint**. We'll call these accounts **UserTokenA** and **UserTokenB** (not pictured in the diagram). The user deposits the tokens they want to swap into **UserTokenA**.
+
+Then, the user creates a throwaway keypair, and approves that keypair to transfer funds from **UserTokenA**. This is covered above:
+
+> Additionally, the user must allow for tokens to be transferred from their source token account. The best practice is to `spl_token::instruction::approve` a precise amount to a new throwaway Keypair, and then have that new Keypair sign the swap transaction. This limits the amount of tokens that can be taken from the user's account by the program.
```

Comment:
nit: Perhaps just a link to the part rather than copy / pasting from earlier in the doc?